pkg/providers/gitlab.go

```diff
@@ -53,7 +40,8 @@ func (g *gitLab) Fetch() (*File, error) {
 	} else {
 		// TODO: handle case when repo doesn't have releases?
 		log.Infof("Getting latest release for %s/%s", g.owner, g.repo)
-		name, _, err := g.GetLatestVersion()
+		var name string
+		name, _, err = g.GetLatestVersion()
```

Comment:
I don't understand this change. Is it about re-assigning an existing variable (`err`)?

Reply:
With `:=` all variables on the left hand side are defined in the respective block, if not yet defined. Therefore, there is a new `err` variable defined. This `err` variable is then re-used on line 48 (`GetRelease`). The check, if the `err` is nil is done outside of the block and therefore checking against the variable defined on line 35 and not against the `err` from the `else` block. It is a combination of shadowing and ineffective assignment.